server/plugin.go

```diff
@@ -51,10 +51,8 @@ type Plugin struct {
 
 	botUserID string
 
-	client *analytics.Client
-
-	// blockSegmentEvents prevents the plugin from sending events to Segment during testing.
-	blockSegmentEvents bool
+	client  telemetry.Client
+	tracker telemetry.Tracker
```

Comment:
Out of scope of this PR: After seeing this API for the first time getting used, I'm wondering what the values of separating the `client` and `tracker` is.

Does it help adapting the strategy if `config.LogSettings.EnableDiagnostics` changes?

Reply:
One of the reasons is to separate responsibilities. The tracker just format the information in a way the client can process it, and add any important extra information. The client just send this information to the telemetry server.
This also helps to expand the functionality. If we want to add any other telemetry client (Google Analytics, Segment, ...) we just need to conform with the client interface, and we do not have to think about "what important information we have to track". That is already done by the tracker. Also it makes way easier to change telemetry clients in case we ever change them again.

Finally, we are creating a new tracker any time the configuration changes, mainly because of the EnableDiagnostics potentially changing. In general, the client has no information that has to be changed, so there is no need to keep opening connections with the client.

Reply:
While the points you made are very valid, it still _feels_ to me that the API is to complex for the simple things we want to do here. But it might come down to "simplicity vs. expendability".

0/5 does it make sense to _not_ store `client` in `Plugin` and re-create it every time 'tracker` gets newly created? That might make the code easier to read and couples these two things closer together.

Reply:
If we were to recreate the client every time we recreate the tracker, we would have to close the previous client (to make sure the events are flushed into the server). On configuration change you would need to do something like:
```
p.tracker.Close()
p.tracker := tracker.New(...)
```
Or expose a function Update (or similar) that would do something like:
```
p.tracker.Update(...)
```

Again, if the tracker and the client are tightly coupled, how can we use a different client if needed? Changing the code of the tracker?